app/tests/spec/models/account.js

```diff
@@ -1307,68 +1307,64 @@ define(function (require, exports, module) {
       it('changes from old to new password', function () {
         var oldPassword = 'password';
         var newPassword = 'new_password';
-        account.set('sessionTokenContext', 'foo');
 
-        sinon.stub(fxaClient, 'checkPassword', function () {
-          return p();
+        account.set({
+          sessionToken: 'sessionToken',
+          sessionTokenContext: 'foo'
         });
 
-        sinon.stub(fxaClient, 'changePassword', function () {
+        sinon.stub(fxaClient, 'checkPassword', function () {
           return p();
         });
 
-        sinon.stub(fxaClient, 'signIn', function () {
-          return p({});
+        sinon.stub(fxaClient, 'changePassword', function () {
+          return p({
+            keyFetchToken: 'new keyFetchToken',
+            sessionToken: 'new sessionToken',
+            uid: 'uid',
+            verified: true
+          });
         });
 
-
         return account.changePassword(oldPassword, newPassword, relier)
           .then(function () {
             assert.isTrue(fxaClient.checkPassword.calledWith(
               EMAIL, oldPassword));
             assert.isTrue(fxaClient.changePassword.calledWith(
-              EMAIL, oldPassword, newPassword));
-            assert.isTrue(fxaClient.signIn.calledWith(
-              EMAIL,
-              newPassword,
-              relier,
-              {
-                reason: SIGN_IN_REASONS.PASSWORD_CHANGE,
-                sessionTokenContext: 'foo'
-              }
-            ));
+              EMAIL, oldPassword, newPassword, 'sessionToken', 'foo', relier));
+
+            assert.equal(account.get('keyFetchToken'), 'new keyFetchToken');
+            assert.equal(account.get('sessionToken'), 'new sessionToken');
+            assert.equal(account.get('sessionTokenContext'), 'foo');
+            assert.equal(account.get('uid'), 'uid');
+            assert.isTrue(account.get('verified'));
           });
       });
     });
 
     describe('completePasswordReset', function () {
-      it('completes the password reset, signs the user in', function () {
+      it('completes the password reset', function () {
```

Comment:
> Add test case for old password reset?

All taken care of in the fxa-client now, [where there is a test](https://github.com/mozilla/fxa-content-server/pull/3747/files/7aaabf562558370995f0929ca31546c8af8c7b3d#diff-c4fce5a20a1f1b633ee88c427c1a7acdR743).